src/mozmlops/cloud_storage_api_client.py

```diff
@@ -52,8 +47,15 @@ def store(self, data: bytes, storage_path: str) -> str:
             # Google recommends setting `if_generation_match=0` if the
             # object is expected to be new. We don't expect collisions,
             # so setting this to 0 seems good.
-            blob.upload_from_file(f, if_generation_match=0)
-            logging.info(f"The model is stored at {storage_path}")
+            try:
```

Comment:
Note that `client.get_bucket` ([docs](https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.client.Client#google_cloud_storage_client_Client_get_bucket)) can raise an exception as well, which we should likely catch

Reply:
It does raise an exception. The reasons I would consider not catching this one are that:

1. If the client does not point to a valid project or bucket, there's no way for storage or fetching to continue, so what we'd do upon catching (unless I am mistaken) is raise another exception.
2. Re-raising allows us to make the error message clearer, but in this case the error looks like: `'404 GET https://storage.googleapis.com/storage/v1/b/mlops-nonexistent-bucket?projection=noAcl&prettyPrint=false: The specified bucket does not exist.'` which is maybe sufficiently clear about the problem that we don't need to provide more details. It's much clearer than the `PreconditionNotMet` exception that I re-raise in `.store()` to clearly state "this means there's already a file in that location." 

@Dexterp37 happy to catch and re-raise this one if you think we should though!

Reply:
No no, your reasoning is sound, maybe I'd just add a one line comment above that line to capture what you said